File: frontend/src/types/calculation.ts
export interface CalculationResult {
  id: string;
  return_id: string;
  calculated_at: string;
  total_income: number;
  agi: number;
  taxable_income: number;
  total_tax: number;
  total_credits: number;
  total_payments: number;
  refund_amount: number;
  amount_owed: number;
  effective_tax_rate: number;
  marginal_tax_rate: number;
  standard_deduction_amount: number;
  itemized_deduction_amount: number;
  deduction_method: string | null;
  form_results: Record<string, Record<string, number>> | null;
  required_forms: string[] | null;
  errors: string[] | null;
  warnings: string[] | null;
}

File: backend/app/schemas/calculation.py
from datetime import datetime

from pydantic import BaseModel


class CalculationResultResponse(BaseModel):
    id: str
    return_id: str
    calculated_at: datetime

    # Summary figures
    total_income: float
    agi: float
    taxable_income: float
    total_tax: float
    total_credits: float
    total_payments: float
    refund_amount: float
    amount_owed: float
    effective_tax_rate: float
    marginal_tax_rate: float

    # Deduction info
    standard_deduction_amount: float
    itemized_deduction_amount: float
    deduction_method: str | None

    # Detailed results
    form_results: dict | None
    required_forms: list[str] | None
    errors: list | None
    warnings: list | None

    model_config = {"from_attributes": True}


class CalculationSummary(BaseModel):
    total_income: float
    agi: float
    taxable_income: float
    total_tax: float
    refund_amount: float
    amount_owed: float
    effective_tax_rate: float
    deduction_method: str | None
